src/main/java/pl/betse/beontime/timesheet/repository/TimeEntryRepository.java

```diff
@@ -0,0 +1,30 @@
+package pl.betse.beontime.timesheet.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import pl.betse.beontime.timesheet.entity.TimeEntryEntity;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Repository
+public interface TimeEntryRepository extends JpaRepository<TimeEntryEntity, Long> {
+
+    List<TimeEntryEntity> findByUserGuidAndWeek(String userGuid, String weekNumber);
+
+    List<TimeEntryEntity> findByUserGuidAndProjectGuidAndWeek(String userGuid, String projectGuid, String weekNumber);
+
+    boolean existsByUserGuidAndProjectGuidAndEntryDate(String userGuid, String projectGuid, LocalDate entryDate);
+
+    boolean existsByProjectGuid(String projectGuid);
+
+    boolean existsByUserGuid(String userGuid);
+
+    @Query("SELECT entry FROM TimeEntryEntity entry WHERE entry.userGuid=:userGuid AND MONTH(entry.entryDate)=MONTH(:localDate) AND YEAR(entry.entryDate)=YEAR(:localDate)")
```

Comment:
using SQL functions inside query in repository is not a good idea - you should prepare data before and ask in very simple query. Problem can occur when you would like to change DB engine

Reply:
I don't know any other solution to obtain rows by YEAR and MONTH from whole date saved in DB.